crates/flipperzero/src/notification/mod.rs

```diff
@@ -0,0 +1,282 @@
+//! Furi dialogs.
+
+#[cfg(feature = "alloc")]
+use alloc::boxed::Box;
+
+use core::ffi::c_char;
+use core::ptr;
+
+use flipperzero_sys as sys;
+use flipperzero_sys::furi::{Status, UnsafeRecord};
+
+///Default notification messages.
+pub mod messages;
+///Default notification notes.
+pub mod notes;
+///Default notification sequences.
+pub mod sequences;
+
+const RECORD_NOTIFICATION: *const c_char = sys::c_string!("notification");
+
+/// A handle to the Notification app.
+pub struct NotificationApp {
+    data: UnsafeRecord<sys::NotificationApp>,
+}
+
+impl NotificationApp {
+    /// Obtains a handle to the Notifications app.
+    pub fn open() -> Self {
+        Self {
+            data: unsafe { UnsafeRecord::open(RECORD_NOTIFICATION) },
+        }
+    }
+
+    /// Runs a notification sequence.
+    #[cfg(feature = "alloc")]
+    pub fn notify<const N: usize>(&mut self, sequence: [&'static NotificationMessage; N]) {
+        unsafe {
+            sys::notification_message(
```

Comment:
I don't believe it's safe to use a stack-allocated `sequence` with `notification_message`.

Internally this is wrapped in a `NotificationAppMessage` and [copied to a message queue](https://sourcegraph.com/github.com/flipperdevices/flipperzero-firmware@0.75.0/-/blob/applications/services/notification/notification_app_api.c?L10), but only the pointer to the sequence is copied, not the actual array. By the time the `NotificationAppMessage` is processed, the stack pointer will no longer be valid.

This shouldn't be a problem with `notification_message_block` because it waits for the notification to be processed before returning.